### PR TITLE
Remove unnecessary check of trailing trivia for directives.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Directives can only be in leading trivia.  So this walk is unnecessary.

### CODE REVIEW

File: src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs [hunk 1]
```diff
@@ -849,10 +849,13 @@ internal IList<TDirective> GetDirectives<TDirective>(Func<TDirective, bool>? fil
...
-            if (node._token != null && node.AsToken() is var token && token.ContainsDirectives)
+            if (node._token != null)
             {
-                GetDirectives(token.LeadingTrivia, filter, ref directives);
-                GetDirectives(token.TrailingTrivia, filter, ref directives);
+                if (node._token.ContainsDirectives)
+                {
+                    foreach (var trivia in node.AsToken().LeadingTrivia)
+                        GetDirectivesInTrivia(trivia, filter, ref directives);
+                }
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs:858]: simplified check and inlined.

File: src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs [hunk 2]
```diff
@@ -865,40 +868,21 @@ private static void GetDirectives<TDirective>(SyntaxNode node, Func<TDirective,
         {
             foreach (var trivia in node.DescendantTrivia(node => node.ContainsDirectives, descendIntoTrivia: true))
             {
-                _ = GetDirectivesInTrivia(trivia, filter, ref directives);
+                GetDirectivesInTrivia(trivia, filter, ref directives);
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs:873]: no need for return value.

File: src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs [hunk 3]
```diff
@@ -865,40 +868,21 @@ private static void GetDirectives<TDirective>(SyntaxNode node, Func<TDirective,
...
-            }
-            return false;
-        }
-
-        private static void GetDirectives<TDirective>(in SyntaxTriviaList trivia, Func<TDirective, bool>? filter, ref List<TDirective>? directives)
-            where TDirective : SyntaxNode
-        {
-            foreach (var tr in trivia)
-            {
-                if (!GetDirectivesInTrivia(tr, filter, ref directives) && tr.GetStructure() is SyntaxNode node)
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs:898]: directives can't be *inside* other structured trivia.  So just removed this.

**Comment by CyrusNajmabadi** [comment]
@333fred @RikkiGibson ptal.

**Review by 333fred — APPROVED** [review]

File: src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs [hunk 4]
```diff
@@ -841,64 +841,50 @@ public bool IsIncrementallyIdenticalTo(SyntaxNodeOrToken other)
...
             {
                 if (trivia.GetStructure() is TDirective directive &&
                     filter?.Invoke(directive) != false)
                 {
-                    if (directives == null)
-                    {
-                        directives = new List<TDirective>();
-                    }
-
+                    directives ??= [];
```
Comment (RikkiGibson) [inline src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs:885]: ✨

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**